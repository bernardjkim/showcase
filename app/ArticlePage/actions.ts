/*
 *
 * ArticlePage actions
 *
 */

import { ArticleActionTypes, SET_ARTICLE_ID } from './types';

export function setArticleId(articleId: string): ArticleActionTypes {
  return {
    type: SET_ARTICLE_ID,
    articleId,
  };
}
