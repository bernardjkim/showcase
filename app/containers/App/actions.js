/*
 * 
 * App actions
 * 
 */

import {
  CREATE_TOKEN,
  CREATE_TOKEN_ERROR,
  CREATE_TOKEN_SUCCESS,
  DELETE_TOKEN,
  DELETE_TOKEN_ERROR,
  DELETE_TOKEN_SUCCESS,
  LOAD_USER,
  LOAD_USER_ERROR,
  LOAD_USER_SUCCESS,
} from './constants';

/**
 * Create token, this action starts the request saga
 *
 * @param  {string}  email    - The user email
 * @param  {string}  password - The user password
 *
 * @return {object}           - An action object with a type of CREATE_TOKEN
 */
export function createToken(email, password) {
  return {
    type: CREATE_TOKEN,
    email,
    password,
  };
}

/**
 * Dispatched when creating the token is successful
 *
 * @return  {object}        - An action object with a type of CREATE_TOKEN_SUCCESSFUL
 */
export function createTokenSuccess() {
  return {
    type: CREATE_TOKEN_SUCCESS,
  };
}

/**
 * Dispatched when creating the token failed
 *
 * @param   {string}  error - Received error
 *
 * @return  {object}        - An action object with a type of CREATE_TOKEN_ERROR
 */
export function createTokenError(error) {
  return {
    type: CREATE_TOKEN_ERROR,
    error,
  };
}

/**
 * Delete token
 *
 * @return  {object}  - An action object with a type of DELETE_TOKEN
 */
export function deleteToken() {
  return {
    type: DELETE_TOKEN,
  };
}

/**
 * Dispatched when deleteing token fails
 *
 * @return  {object}  - An action object with a type of DELETE_TOKEN_SUCCESS
 */
export function deleteTokenSuccess() {
  return {
    type: DELETE_TOKEN_SUCCESS,
  };
}

/**
 * Dispatched when deleteing token is successful
 *
 * @param   {Error} error - Error received
 *
 * @return  {object}      - An action object with a type of DELETE_TOKEN_SUCCESS
 */
export function deleteTokenError(error) {
  return {
    type: DELETE_TOKEN_ERROR,
    error,
  };
}

/**
 * Load user, this action start the request saga
 *
 * @return  {object}  - An action object with a type of LOAD_USER
 */
export function loadUser() {
  return {
    type: LOAD_USER,
  };
}

/**
 * Dispatched when loading user is successful
 *
 * @param   {User}  user  - Received user
 *
 * @return  {object}      - An action object with a type of LOAD_USER_SUCCESS
 */
export function loadUserSuccess(user) {
  return {
    type: LOAD_USER_SUCCESS,
    user,
  };
}

/**
 * Dispatched when loading user fails
 *
 * @param   {Error} error - Error received
 *
 * @return  {object}      - An action object with a type of LOAD_USER_ERROR
 */
export function loadUserError(error) {
  return {
    type: LOAD_USER_ERROR,
    error,
  };
}
