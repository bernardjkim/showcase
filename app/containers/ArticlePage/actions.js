/*
 *
 * ArticlePage actions
 *
 */

import {
  CREATE_COMMENT,
  CREATE_COMMENT_ERROR,
  CREATE_COMMENT_SUCCESS,
  LIKE_ARTICLE,
  LIKE_ARTICLE_ERROR,
  LIKE_ARTICLE_SUCCESS,
  LOAD_ARTICLE,
  LOAD_ARTICLE_ERROR,
  LOAD_ARTICLE_SUCCESS,
  LOAD_COMMENTS,
  LOAD_COMMENTS_ERROR,
  LOAD_COMMENTS_SUCCESS,
} from './constants';

/**
 * Create comment, this action starts the request saga
 *
 * @param  {object}  comment  - The comment value
 *
 * @return {object}           - An action object with a type of CREATE_COMMENT
 */
export function createComment(comment) {
  return {
    type: CREATE_COMMENT,
    comment,
  };
}

/**
 * Dispatched when posting the comment is acknowledged by the request saga
 *
 * @param  {object}  comment  - The comment data
 *
 * @return {object}           - An action object with a type of CREATE_COMMENT_SUCCESS
 */
export function createCommentSuccess(comment) {
  return {
    type: CREATE_COMMENT_SUCCESS,
    comment,
  };
}

/**
 * Dispatched when posting the comment fails
 *
 * @return {object} - An action object with a type of CREATE_COMMENT_SUCCESS
 */
export function createCommentError(error) {
  return {
    type: CREATE_COMMENT_ERROR,
    error,
  };
}

/**
 * Like article, this action starts the request saga
 *
 * @return {object} - An action object with a type of LIKE_ARTICLE
 */
export function likeArticle() {
  return {
    type: LIKE_ARTICLE,
  };
}

/**
 * Dispatched when posting the like is acknowledged by the request saga
 *
 * @return {object} - An action object with a type of LIKE_ARTICLE_SUCCESS
 */
export function likeArticleSuccess() {
  return {
    type: LIKE_ARTICLE_SUCCESS,
  };
}

/**
 * Dispatched when posting the like fails
 *
 * @return {object} - An action object with a type of LIKE_ARTICLE_ERROR
 */
export function likeArticleError(error) {
  return {
    type: LIKE_ARTICLE_ERROR,
    error,
  };
}

/**
 * Load article, this action starts the request saga
 *
 * @param  {object}  query  - The comment data
 *
 * @return {object}         - An action object with a type of LOAD_ARTICLE
 */
export function loadArticle(query) {
  return {
    type: LOAD_ARTICLE,
    query,
  };
}

/**
 * Dispatched when the article is loaded by the request saga
 *
 * @param  {object}  article  - The article data
 *
 * @return {object}           - An action object with a type of LOAD_ARTICLE_SUCCESS
 */
export function loadArticleSuccess(article) {
  return {
    type: LOAD_ARTICLE_SUCCESS,
    article,
  };
}

/**
 * Dispatched when loading the article fails
 *
 * @param  {object} error The error
 *
 * @return {object}       An action object with a type of LOAD_ARTICLE_ERROR
 */
export function loadArticleError(error) {
  return {
    type: LOAD_ARTICLE_ERROR,
    error,
  };
}

/**
 * Load comments, this action starts the request saga
 *
 * @return {object} - An action object with a type of LOAD_COMMENTS
 */
export function loadComments() {
  return {
    type: LOAD_COMMENTS,
  };
}

/**
 * Dispatched when the comments are loaded by the request saga
 *
 * @param  {array}  comments  - The comments data
 *
 * @return {object}           - An action object with a type of LOAD_COMMENT_SUCCESS
 */
export function loadCommentsSuccess(comments) {
  return {
    type: LOAD_COMMENTS_SUCCESS,
    comments,
  };
}

/**
 * Dispatched when loading the comments fails
 *
 * @param  {object} error - The error
 *
 * @return {object}       - An action object with a type of LOAD_COMMENTS_ERROR
 */
export function loadCommentsError(error) {
  return {
    type: LOAD_COMMENTS_ERROR,
    error,
  };
}
