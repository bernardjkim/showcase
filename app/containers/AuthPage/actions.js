/*
 *
 * AuthPage actions
 *
 */

import {
  CLEAR_ERRORS,
  CREATE_TOKEN,
  CREATE_TOKEN_ERROR,
  CREATE_TOKEN_SUCCESS,
  CREATE_USER,
  CREATE_USER_ERROR,
  CREATE_USER_SUCCESS,
} from './constants';

/**
 * Clear error messages
 *
 * @return  {object}  - An action object with a type of CLEAR_ERRORS
 */
export function clearErrors() {
  return {
    type: CLEAR_ERRORS,
  };
}

/**
 * Create token, this action starts the request saga
 *
 * @param  {string}  email    - The user email
 * @param  {string}  password - The user password
 *
 * @return {object}           - An action object with a type of CREATE_TOKEN
 */
export function createToken(email, password) {
  return {
    type: CREATE_TOKEN,
    email,
    password,
  };
}

/**
 * Dispatched when creating the token is successful
 *
 * @return  {object}        - An action object with a type of CREATE_TOKEN_SUCCESSFUL
 */

export function createTokenSuccess() {
  return {
    type: CREATE_TOKEN_SUCCESS,
  };
}

/**
 * Dispatched when creating the token failed
 *
 * @param   {string}  error - Received error
 *
 * @return  {object}        - An action object with a type of CREATE_TOKEN_ERROR
 */
export function createTokenError(error) {
  return {
    type: CREATE_TOKEN_ERROR,
    error,
  };
}
export function createUser(username, email, password, passwordConfirm) {
  return {
    type: CREATE_USER,
    username,
    email,
    password,
    passwordConfirm,
  };
}

export function createUserSuccess() {
  return {
    type: CREATE_USER_SUCCESS,
  };
}

export function createUserError(error) {
  return {
    type: CREATE_USER_ERROR,
    error,
  };
}
