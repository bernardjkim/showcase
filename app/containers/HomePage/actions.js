/*
 *
 * HomePage actions
 *
 */

import {
  LOAD_ARTICLES_ALL,
  LOAD_ARTICLES_ALL_ERROR,
  LOAD_ARTICLES_ALL_SUCCESS,
} from './constants';

/**
 * Load articles all, this action starts the request saga
 *
 * @return {object} - An action object with a type of LOAD_ARTICLES_ALL
 */
export function loadArticlesAll() {
  return {
    type: LOAD_ARTICLES_ALL,
  };
}

/**
 * Dispatched when loading articles all is successful
 *
 * @param   {Articles[]}  articles  - Array of articles
 *
 * @returns {object}                - An action object with a type of LOAD_ARTICLES_ALL_SUCCESS
 */
export function loadArticlesAllSuccess(articles) {
  return {
    type: LOAD_ARTICLES_ALL_SUCCESS,
    articles,
  };
}

/**
 * Dispatched when loading articles fails
 *
 * @param   {Error}   error - Received error
 *
 * @returns {object}        - An action object with a type of LOAD_ARTICLES_ALL_ERROR
 */
export function loadArticlesAllError(error) {
  return {
    type: LOAD_ARTICLES_ALL_ERROR,
    error,
  };
}
