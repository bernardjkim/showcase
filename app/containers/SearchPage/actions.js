/*
 *
 * SearchPage actions
 *
 */

import {
  LOAD_ARTICLES,
  LOAD_ARTICLES_ERROR,
  LOAD_ARTICLES_SUCCESS,
  LOAD_ARTICLES_ALL,
  LOAD_ARTICLES_ALL_ERROR,
  LOAD_ARTICLES_ALL_SUCCESS,
} from './constants';

/**
 * Load articles, this action starts the request saga
 *
 * @param  {object}  query  - The query object
 *
 * @return {object}         - An action object with a type of LOAD_ARTICLES
 */
export function loadArticles(query) {
  return {
    type: LOAD_ARTICLES,
    query,
  };
}

/**
 * Dispatched when loading articles is successful
 *
 * @param   {Articles[]}  articles  - Array of articles
 *
 * @returns {object}                - An action object with a type of LOAD_ARTICLES_SUCCESS
 */
export function loadArticlesSuccess(articles) {
  return {
    type: LOAD_ARTICLES_SUCCESS,
    articles,
  };
}

/**
 * Dispatched when loading articles fails
 *
 * @param   {Error}   error - Received error
 *
 * @returns {object}        - An action object with a type of LOAD_ARTICLES_ERROR
 */
export function loadArticlesError(error) {
  return {
    type: LOAD_ARTICLES_ERROR,
    error,
  };
}
