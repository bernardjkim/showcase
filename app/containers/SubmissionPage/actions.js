/*
 *
 * SubmissionPage actions
 *
 */

import {
  SUBMIT_FORM,
  SUBMIT_FORM_ERROR,
  SUBMIT_FORM_SUCCESS,
} from './constants';

export function submitForm(form) {
  return {
    type: SUBMIT_FORM,
    form,
  };
}

export function submitFormSuccess() {
  return {
    type: SUBMIT_FORM_SUCCESS,
  };
}

export function submitFormError(error) {
  return {
    type: SUBMIT_FORM_ERROR,
    error,
  };
}
