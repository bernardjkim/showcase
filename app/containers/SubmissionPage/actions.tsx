/*
 *
 * SubmissionPage actions
 *
 */

import {
  Form,
  Error,
  CLEAR_STATE,
  SUBMIT_FORM,
  SUBMIT_FORM_ERROR,
  SUBMIT_FORM_SUCCESS,
  SubmissionActionTypes,
} from './types';

export function clearState(): SubmissionActionTypes {
  return {
    type: CLEAR_STATE,
  };
}

export function submitForm(form: Form): SubmissionActionTypes {
  return {
    type: SUBMIT_FORM,
    form,
  };
}

export function submitFormSuccess(): SubmissionActionTypes {
  return {
    type: SUBMIT_FORM_SUCCESS,
  };
}

export function submitFormError(error: Error): SubmissionActionTypes {
  return {
    type: SUBMIT_FORM_ERROR,
    error,
  };
}
