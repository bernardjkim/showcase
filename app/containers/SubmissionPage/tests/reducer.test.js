import { fromJS } from 'immutable';
import submissionPageReducer from '../reducer';

import { submitForm, submitFormError, submitFormSuccess } from '../actions';

describe('submissionPageReducer', () => {
  let state;
  beforeEach(() => {
    state = fromJS({
      loading: false,
      error: false,
      submissionSuccess: false,
    });
  });
  it('returns the initial state', () => {
    expect(submissionPageReducer(undefined, {})).toMatchSnapshot();
  });

  it('handle the submit form action', () => {
    expect(submissionPageReducer(state, submitForm({}))).toMatchSnapshot();
  });

  it('handle the submit form success action', () => {
    expect(submissionPageReducer(state, submitFormSuccess())).toMatchSnapshot();
  });

  it('handle the submit form error action', () => {
    expect(submissionPageReducer(state, submitFormError({}))).toMatchSnapshot();
  });
});
