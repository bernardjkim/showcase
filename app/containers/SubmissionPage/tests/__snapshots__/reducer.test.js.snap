// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`submissionPageReducer handle the submit form action 1`] = `
Immutable.Map {
  "loading": true,
  "error": false,
  "submissionSuccess": false,
}
`;

exports[`submissionPageReducer handle the submit form error action 1`] = `
Immutable.Map {
  "loading": false,
  "error": Immutable.Map {},
  "submissionSuccess": false,
}
`;

exports[`submissionPageReducer handle the submit form success action 1`] = `
Immutable.Map {
  "loading": false,
  "error": false,
  "submissionSuccess": true,
}
`;

exports[`submissionPageReducer returns the initial state 1`] = `
Immutable.Map {
  "loading": false,
  "error": false,
  "submissionSuccess": false,
}
`;
